GDJS/Runtime/pixi-renderers/layer-pixi-renderer.js

```diff
@@ -252,59 +257,101 @@ gdjs.LayerPixiRenderer.prototype.enableEffect = function(name, value) {
+    }
+    this._oldWidth = this._pixiRenderer.screen.width;
+    this._oldHeight = this._pixiRenderer.screen.height;
   }
 
   var oldRenderTexture = this._pixiRenderer.renderTexture.current;
   var oldSourceFrame = this._pixiRenderer.renderTexture.sourceFrame;
 
   this._pixiRenderer.renderTexture.bind(this._renderTexture);
-  this._pixiRenderer.renderTexture.clear([0.25, 0.25, 0.25, 1.0]);
+  //TODO: Pass data for clear color.
+  this._pixiRenderer.renderTexture.clear([0.5, 0.5, 0.5, 1.0]);
 
   this._pixiRenderer.render(this._pixiContainer, this._renderTexture, false);
-  this._pixiRenderer.renderTexture.bind(oldRenderTexture, oldSourceFrame, undefined);
-}
+  this._pixiRenderer.renderTexture.bind(
+    oldRenderTexture,
+    oldSourceFrame,
+    undefined
+  );
+};
 
-gdjs.LayerPixiRenderer.prototype.getRenderTexture = function() {
-  if(!this._renderTexture) this.updateRenderTexture();
+gdjs.LayerPixiRenderer.prototype.getRenderTexture = function () {
+  if (!this._renderTexture) this.updateRenderTexture();
   return this._renderTexture;
-}
+};
 
-gdjs.LayerPixiRenderer.prototype.addLayerToLighting = function() {
-  if(!this._lightingSprite) {
+gdjs.LayerPixiRenderer.prototype.addLayerToLighting = function () {
+  if (!this._lightingSprite) {
     this._lightingSprite = new PIXI.Sprite(this.getRenderTexture());
     this._lightingSprite.blendMode = PIXI.BLEND_MODES.MULTIPLY;
-    this._runtimeSceneRenderer.getPIXIContainer().addChild(this._lightingSprite);
+    // fix for blend mode when applying filter
+    if (this._pixiContainer.filters) {
+      this._pixiContainer.filterArea = new PIXI.Rectangle(
+        0,
+        0,
+        this._pixiRenderer.screen.width,
+        this._pixiRenderer.screen.height
+      );
+      for (var i = 0; i < this._pixiContainer.filters.length; i++) {
+        this._pixiContainer.filters[i].blendMode = PIXI.BLEND_MODES.ADD;
+      }
+    }
+    this._runtimeSceneRenderer
+      .getPIXIContainer()
+      .addChild(this._lightingSprite);
   }
-}
+};
 
-gdjs.LayerPixiRenderer.prototype.getPIXIContainer = function() {
+gdjs.LayerPixiRenderer.prototype.getPIXIContainer = function () {
   return this._pixiContainer;
-}
\ No newline at end of file
+};
+
+gdjs.LayerPixiRenderer.prototype.syncWithBaseLayer = function () {
+  if (!this._baseLayer) {
+    this._baseLayer = this._runtimeSceneRenderer.getScene().getLayer('');
+    this._baseLayerCameraX = this._baseLayer.getCameraX();
+    this._baseLayerCameraY = this._baseLayer.getCameraY();
+  }
+  if (
+    this._baseLayer &&
+    (this._baseLayerCameraX !== this._baseLayer.getCameraX() ||
+      this._baseLayerCameraY !== this._baseLayer.getCameraY())
+  ) {
+    this._layer.setCameraX(this._baseLayer.getCameraX());
+    this._layer.setCameraY(this._baseLayer.getCameraY());
+    this._baseLayerCameraX = this._baseLayer.getCameraX();
+    this._baseLayerCameraY = this._baseLayer.getCameraY();
+  }
+};
```

Comment:
I'm not sure about this, we are checking whether the base layer moved or not every frame. We could simply add an action to change the camera of lights layer (when we change base layer's camera). I don't think the performance hit we get due to the overhead of checks is really justified.
What do you think @4ian? Should we even do this automatically or not? If yes, do you think we could have a better way of doing it?

Reply:
> Should we even do this automatically or not? 

I think yes, at least we should have an option "automatically follow the base layer" :) 
So that when we're creating a lighting layer we activate:
- "automatically follow the base layer" 
- the render texture
- a default kawase blur filter (much faster than the traditional blur) + automatically use blend mode add on the filter.


>  If yes, do you think we could have a better way of doing it?

I think we can just do a `this._layer.setCameraX(this._baseLayer.getCameraX());` (in gdjs.Layer, not necessarily in the pixi renderer) (this "automatically follow the base layer" option should be something independent from pixi).
No need to store the layer in this._baseLayer nor to do the checks against the X/Y position, because:
- the checks will be more costly than just doing the assignment.
- it's probable that the X/Y position are changed every frame anyway :)

Reply:
Can you explain a bit more about doing stuff in gdjs.Layer? Do you mean I should move the `syncWithBaseLayer` funtion to gdjs.Layer instead? 

> No need to store the layer in this._baseLayer

I still need some reference to the base layer, right? Even if I move it to gdjs.Layer I need to have some reference to base layer, no?

Reply:
> Do you mean I should move the syncWithBaseLayer funtion to gdjs.Layer instead

Yes that's my idea. Having a layer automatically following the base layer is a feature that is not tied to the rendering engine, so I think it should be in gdjs.Layer - the PIXI renderer is just "reading" into gdjs.Layer to do its job of rendering the layer.

>  Even if I move it to gdjs.Layer I need to have some reference to base layer, no?

You can fetch it using `getLayer("")` at every frame I think?

Reply:
Note that the base layer is supposed to be always existing (by convention) and there is no reason for it to be changed once created, so keeping a reference to it is ok - but calling `getLayer("")` should be very fast (it's looking into a hashtable aka a JS object) so it's ok to make this call - so that in the future if for some reason the reference to the layer changes (you never know), we're safe.